idea/idea-fir/src/org/jetbrains/kotlin/idea/core/overrideImplement/KtOverrideMembersHandler.kt

```diff
@@ -0,0 +1,99 @@
+
+import org.jetbrains.kotlin.descriptors.Modality
+import org.jetbrains.kotlin.idea.core.util.KotlinIdeaCoreBundle
+import org.jetbrains.kotlin.idea.frontend.api.KtAnalysisSession
+import org.jetbrains.kotlin.idea.frontend.api.analyse
+import org.jetbrains.kotlin.idea.frontend.api.symbols.KtCallableSymbol
+import org.jetbrains.kotlin.idea.frontend.api.symbols.KtClassKind
+import org.jetbrains.kotlin.idea.frontend.api.symbols.KtClassOrObjectSymbol
+import org.jetbrains.kotlin.idea.frontend.api.symbols.markers.KtSymbolWithModality
+import org.jetbrains.kotlin.idea.frontend.api.tokens.HackToForceAllowRunningAnalyzeOnEDT
+import org.jetbrains.kotlin.idea.frontend.api.tokens.hackyAllowRunningOnEdt
+import org.jetbrains.kotlin.idea.frontend.api.types.KtClassType
+import org.jetbrains.kotlin.name.StandardClassIds
+import org.jetbrains.kotlin.psi.KtClassOrObject
+
+class KtOverrideMembersHandler(private val preferConstructorParameters: Boolean = false) : KtGenerateMembersHandler() {
+    @OptIn(HackToForceAllowRunningAnalyzeOnEDT::class)
+    override fun collectMembersToGenerate(classOrObject: KtClassOrObject): Collection<KtClassMember> {
+        return hackyAllowRunningOnEdt {
+            analyse(classOrObject) {
+                collectMembers(classOrObject)
+            }
+        }
+    }
+
+    @OptIn(ExperimentalStdlibApi::class)
+    private fun KtAnalysisSession.collectMembers(classOrObject: KtClassOrObject): List<KtClassMember> {
+        val classOrObjectSymbol = classOrObject.getClassOrObjectSymbol()
+        return buildList {
+            classOrObjectSymbol.processMemberSymbols { symbol, _, _, isOverridable, intersectionSymbols ->
+                if (!isOverridable) return@processMemberSymbols
+
+                val symbolsToProcess = if (intersectionSymbols?.isNotEmpty() == true) {
+                    val nonAbstractMembers = intersectionSymbols.filter { (it as? KtSymbolWithModality)?.modality != Modality.ABSTRACT }
+                    // If there are non-abstract members, we only want to show override for these non-abstract members. Otherwise, show any
+                    // abstract member to override.
+                    nonAbstractMembers.ifEmpty {
+                        listOf(intersectionSymbols.first())
+                    }
+                } else {
+                    listOf(symbol)
+                }
+
+                for (symbolToProcess in symbolsToProcess) {
+                    val originalOverriddenSymbol = symbolToProcess.originalOverriddenSymbol
+                    val containingSymbol = originalOverriddenSymbol?.originalContainingClassForOverride
+
+                    val bodyType = when {
+                        classOrObjectSymbol.classKind == KtClassKind.INTERFACE && containingSymbol?.classIdIfNonLocal == StandardClassIds.Any -> {
+                            if ((classOrObjectSymbol.superTypes.singleOrNull()?.type as? KtClassType)?.classId == StandardClassIds.Any) {
```

Comment:
Let's add `KtTypeInfoProviderMixIn.isAny` like it is done for other builtin types